### ISSUE TITLE
Error: `generateNodeEnvModulesPattern is not a function` with custom Node env

### ISSUE THREAD

**Issue by piotrkulpinski** [teambit/bit#8327]
Hi, 

I've created a custom Node Env in my workspace based on the default `bitdev.node/node-env` with the following command: 

```bash
bit create node-env node --aspect bitdev.node/node-env --scope curiousleaf.envs
```

Now, if I set this env for a component, I can no longer test it correctly due to the following error: 

```bash
✖ service "tester" of env "curiousleaf.envs/node" has failed. error: generateNodeEnvModulesPattern is not a function
```

Any ideas why this might be happening?

### Specifications
- Bit version: 1.5.29
- Workspace type: (harmony/legacy) harmony
- Node version: 20.9.0
- pnpm version: 8.12.1
- Platform: MacOS

**Comment by benjgil** [comment]
Can you please run the same command with --log=error and then paste the error stack trace?

**Comment by piotrkulpinski** [comment]
```bash
err: {
      "type": "TypeError",
      "message": "generateNodeEnvModulesPattern is not a function",
      "stack":
          TypeError: generateNodeEnvModulesPattern is not a function
              at Object.<anonymous> (/.../Sites/curiousleaf-design/node_modules/.pnpm/@bitdev+node.node-env@1.0.6_@teambit+legacy@node_modules+@teambit+legacy_@testing-library+rea_pemsownk6rjdmc7hjsg2v2gosu/node_modules/@bitdev/node.node-env/dist/config/jest.config.js:24:60)
              at Module._compile (node:internal/modules/cjs/loader:1376:14)
              at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
              at Module.load (node:internal/modules/cjs/loader:1207:32)
              at Module._load (node:internal/modules/cjs/loader:1023:12)
              at module.constructor.require (/.../.bvm/versions/1.5.29/bit-1.5.29/node_modules/@teambit/bit/dist/hook-require.js:27:31)
              at require (node:internal/modules/helpers:176:18)
              at JestTester.<anonymous> (/.../Sites/curiousleaf-design/node_modules/.pnpm/@teambit+defender.jest-tester@1.0.13_@teambit+legacy@node_modules+@teambit+legacy_@types+node@12.20.4/node_modules/@teambit/defender.jest-tester/dist/jest-tester.js:155:32)
              at Generator.next (<anonymous>)
              at /.../Sites/curiousleaf-design/node_modules/.pnpm/@teambit+defender.jest-tester@1.0.13_@teambit+legacy@node_modules+@teambit+legacy_@types+node@12.20.4/node_modules/@teambit/defender.jest-tester/dist/jest-tester.js:8:71
              at new Promise (<anonymous>)
              at __awaiter (/.../Sites/curiousleaf-design/node_modules/.pnpm/@teambit+defender.jest-tester@1.0.13_@teambit+legacy@node_modules+@teambit+legacy_@types+node@12.20.4/node_modules/@teambit/defender.jest-tester/dist/jest-tester.js:4:12)
              at JestTester.test (/.../Sites/curiousleaf-design/node_modules/.pnpm/@teambit+defender.jest-tester@1.0.13_@teambit+legacy@node_modules+@teambit+legacy_@types+node@12.20.4/node_modules/@teambit/defender.jest-tester/dist/jest-tester.js:120:16)
              at TesterService.run (/.../.bvm/versions/1.5.29/bit-1.5.29/node_modules/@teambit/tester/dist/tester.service.js:176:34)
              at async /.../.bvm/versions/1.5.29/bit-1.5.29/node_modules/@teambit/envs/dist/runtime/runtime.js:97:31
              at async pMapSeries (/.../.bvm/versions/1.5.29/bit-1.5.29/node_modules/p-map-series/index.js:9:15)
    }
teambit.envs/envs, service "tester" of env "curiousleaf.envs/node" has failed. error: generateNodeEnvModulesPattern is not a function
✖ service "tester" of env "curiousleaf.envs/node" has failed. error: generateNodeEnvModulesPattern is not a function
```

**Comment by MagnesiumDev** [comment]
same thing happening to me

**Comment by benjgil** [comment]
@MagnesiumDev 
Thanks for pointing this out - the pointers are indeed not portable between envs, and the env template should have its own local file references.
@piotrkulpinski that's also what's causing your issue. The jest config in your custom env was actually correct, but as the env didnt have its own internal config paths it was resolving to the config from the base NodeEnv, which has a bug i'm just pushing a fix for.

Thanks to both of you for reporting

**Comment by benjgil** [comment]
This has been fixed from v 1.0.15 of bitdev.node/node-env 
closing
